### PR TITLE
fix: allow the formbodyhandler to run tasks in the calling thread

### LINKED ISSUE #25687: A java.lang.NullPointerException occurs when sending a Multipart/form-data request to any file upload interface.
### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

Whether it is a plug-in or a file upload operation in the admin api, the following error will occur.
## The test interface is /admin/realms/Test/localization/zh_cn
**The call stack is as follows：**
2023-12-19 11:39:13,674 ERROR [org.jboss.threads.errors] (executor-thread-21) Thread Thread[executor-thread-21,5,main] threw an uncaught exception: java.lang.NullPointerException: Cannot invoke "java.util.concurrent.Executor.execute(java.lang.Runnable)" because the return value of "java.util.function.Supplier.get()" is null
        at org.jboss.resteasy.reactive.server.core.multipart.MultiPartParserDefinition$MultiPartUploadHandler.deleteFiles(MultiPartParserDefinition.java:377)
        at org.jboss.resteasy.reactive.server.core.multipart.MultiPartParserDefinition$MultiPartUploadHandler.close(MultiPartParserDefinition.java:371)
        at org.jboss.resteasy.reactive.server.core.multipart.MultiPartParserDefinition$1.onComplete(MultiPartParserDefinition.java:86)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.onComplete(AbstractResteasyReactiveContext.java:389)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.close(AbstractResteasyReactiveContext.java:115)
        at org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext.close(ResteasyReactiveRequestContext.java:364)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:196)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)

### Version

23.0.3

### Expected behavior

No unexpected exceptions occur when processing resource recovery for Multipart requests

### Actual behavior

Exception occurs during Multipart resource recovery

### How to Reproduce?

```python
import requests

realms= Your realms
url = f'http://127.0.0.1:8080/admin/realms/{realms}/localization/zh_cn'
files = {'file': ('example.json', open('example.json', 'rb'), 'text/plain')}
headers = {
    'Authorization': 'your token',
}
response = requests.post(url, files=files, headers=headers)
print(response.text)
```

### Anything else?

1. Test system environment: Microsoft Windows [Version 10.0.19045.3803]
2. keycloak installation path: G:\work\keycloak-23.0.3\keycloak-23.0.3

### PR DESCRIPTION
**Pull request by shawkins**
closes: #25687

@pedroigor @geoand it appears that an executor is required to handle the delete tasks, but from the original comment on KeycloakHandlerChainCustomizer this doesn't seem expected. Is there an upstream issue needed here as well? And does it seem sufficient to just run in the calling thread?

### CODE REVIEW

**Comment by geoand** [comment]
> Is there an upstream issue needed here as well

That's hard to tell as I've never seen such an issue upstream

**Review by vmuzikar — COMMENTED** [review]
@shawkins The fix seems reasonable to me but I wonder why we suddenly run on event-loop thread (as it was a non-blocking code) instead of worker thread.

I also wonder if we're missing some test case for uploads that would reliably catch this. WDYT?

**Review by vmuzikar — APPROVED** [review]
@shawkins Ok, LGTM.

Feel free to merge, or wait for a review from @pedroigor / @geoand.

**Pull request merged**
